### ISSUE TITLE
victorialogs:  some logs are missing when using `sort ... limit`

### ISSUE THREAD

**Issue by francisye19** [VictoriaMetrics/VictoriaMetrics#6321]
### Describe the bug

I ingested 10 rows of log, with field `n` from 1 to 10:
```shell
echo '{"stream": "test_sort_limit", "n": 1, "_msg":"10"}' | curl -XPOST -H 'Content-Type: application/stream+json' --data-binary @- 'http://localhost:9428/insert/jsonline?_stream_fields=stream'
```

**query**: `curl -s http://localhost:9428/select/logsql/query --data-urlencode query='_stream:{stream="test_sort_limit"} | fields n | sort by (n) limit 50'`
**expected**: return 10 rows
**actual**: return only 8 rows, with n from 1 to 8

**Please take a look at** [this line](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/v0.8.0-victorialogs/lib/logstorage/pipe_topk.go#L261). I think the correct condition to return is `uint64(len(rows)) == shard.ps.offset+shard.ps.limit && !topkLess(shard.ps, r, rows[0])`

### To Reproduce

as described above

### Version

v0.8.0

**Comment by Haleygo** [comment]
Dup with https://github.com/VictoriaMetrics/VictoriaMetrics/issues/6298

**Comment by francisye19** [comment]
fixed in v0.9.1
